lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -617,30 +677,42 @@ monitorStakePools tr gp nl DBLayer{..} =
                 liftIO $ traceWith tr $ MsgStakePoolRetirement cert
                 putPoolRetirement publicationTime cert
 
+
+-- | Worker thread that monitors pool metadata and syncs it to the database.
 monitorMetadata
     :: Tracer IO StakePoolLog
     -> GenesisParameters
-    -> (   PoolId
-        -> StakePoolMetadataUrl
-        -> StakePoolMetadataHash
-        -> IO (Maybe StakePoolMetadata)
-       )
     -> DBLayer IO
     -> IO ()
-monitorMetadata tr gp fetchMetadata DBLayer{..} = forever $ do
-    refs <- atomically (unfetchedPoolMetadataRefs 100)
-    successes <- fmap catMaybes $ forM refs $ \(pid, url, hash) -> do
-        fetchMetadata pid url hash >>= \case
-            Nothing -> Nothing <$ do
-                atomically $ putFetchAttempt (url, hash)
-
-            Just meta -> Just hash <$ do
-                atomically $ putPoolMetadata hash meta
-
-    when (null refs || null successes) $ do
-        traceWith tr $ MsgFetchTakeBreak blockFrequency
-        threadDelay blockFrequency
+monitorMetadata tr gp DBLayer{..} = do
+    settings <- atomically readSettings
+    forever $ do
+        (refs, successes) <- case poolMetadataSource settings of
+            FetchNone -> pure ([], [])
+            FetchDirect -> (fetcher [identityUrlBuilder]) >>= fetchThem
+            FetchSMASH (unSmashServer -> uri) -> (fetcher [registryUrlBuilder uri]) >>= fetchThem
+
+        when (null refs || null successes) $ do
+            traceWith tr $ MsgFetchTakeBreak blockFrequency
+            threadDelay blockFrequency
   where
+    fetcher fetchStrategies =
+        fetchFromRemote trFetch fetchStrategies
+            <$> newManager defaultManagerSettings
```

Comment:
Does this create a new manager for every fetch :thinking: ? Wouldn't it be better to factor out the creation of the network manager outside of the forever loop and re-use it on each loop?

Reply:
![Screenshot from 2020-10-01 11-07-16](https://user-images.githubusercontent.com/5680256/94790276-56dada00-03d6-11eb-94d7-72351b553c4d.png)